.github/workflows/continuous-deployment.yml

```diff
@@ -0,0 +1,223 @@
+
+      - uses: actions/checkout@ec3a7ce113134d7a93b817d10a8272cb61118579 # renovate: tag=v2
+        with:
+          ref: ${{ matrix.deployment.sha }}
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Login to DockerHub
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKER_HUB_USERNAME }}
+          password: ${{ secrets.DOCKER_HUB_PASSWORD }}
+
+      - name: Build and push frontend docker image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          file: .docker-hub/frontend/Dockerfile
+          tags: ${{ ((matrix.deployment.name == 'dev') && 'ecamp/ecamp3-frontend:latest,' || '') }}ecamp/ecamp3-frontend:${{ matrix.deployment.sha }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Build and push api docker image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          file: api/Dockerfile
+          tags: ${{ ((matrix.deployment.name == 'dev') && 'ecamp/ecamp3-api:latest,' || '') }}ecamp/ecamp3-api:${{ matrix.deployment.sha }}
+          context: './api'
+          target: api_platform_php
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Build and push caddy docker image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          file: api/Dockerfile
+          tags: ${{ ((matrix.deployment.name == 'dev') && 'ecamp/ecamp3-caddy:latest,' || '') }}ecamp/ecamp3-caddy:${{ matrix.deployment.sha }}
+          context: './api'
+          target: api_platform_caddy_prod
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Build and push print docker image
+        uses: docker/build-push-action@v2
+        with:
+          push: true
+          file: .docker-hub/print/Dockerfile
+          tags: ${{ ((matrix.deployment.name == 'dev') && 'ecamp/ecamp3-print:latest,' || '') }}ecamp/ecamp3-print:${{ matrix.deployment.sha }}
+          cache-from: type=gha
+          cache-to: type=gha,mode=max
+
+      - name: Upgrade or install helm release
+        run: |
+          # Setup authentication
+          mkdir ~/.kube && echo '${{ secrets.KUBECONFIG }}' > ~/.kube/config && chmod go-r ~/.kube/config
+          # Switch to the helm chart directory
+          cd .helm/ecamp3
+          # Install dependency charts
+          helm dependency update
+          # Set the appVersion, workaround from https://github.com/helm/helm/issues/8194 so that we can
+          # later find out which deployments need to be upgraded
+          sed -i 's/^appVersion:.*$/appVersion: "${{ matrix.deployment.sha }}"/' Chart.yaml
+          # Install or upgrade the release
+          helm upgrade --install ecamp3-${{ matrix.deployment.name }} . \
+            --set imageTag=${{ matrix.deployment.sha }} \
+            --set sharedCookieDomain=.ecamp3.ch \
```

Comment:
Doesn't that lead to cookie conflicts across the various deployments? How about we use the deployment name as subdomain namespace:
```
            --set sharedCookieDomain=.${{ matrix.deployment.name }}.ecamp3.ch \
            --set api.domain=api.${{ matrix.deployment.name }}.ecamp3.ch \
            --set frontend.domain=${{ matrix.deployment.name }}.ecamp3.ch \
            --set print.domain=print.${{ matrix.deployment.name }}.ecamp3.ch \
            --set mail.domain=mail.${{ matrix.deployment.name }}.ecamp3.ch \
```

Reply:
I'd love to, but the free version of cloudflare does not support sub-subdomains

Reply:
Hmm, ok, that's a pitty. Just ran into the same issue --> seems the DNS works fine with sub-subdomains, but the SSL certificate cannot handle it.

Reply:
Ah if it's just that, we could also set up automatic ssl certificates ourselves with cert-manager, there's a blog post by digitalocean that explains how

Reply:
As long as we want to use Cloudflare as a proxy, wouldn't we run into the same issue, as Cloudflare would want to issue it's own certificate between visitor and cloudflare server? We would only solve the encryption/connection between server & cloudflare.
Or does Coudflare pass through the certificates in "Full (strict)" mode?

Reply:
Looks like that isn't possible either on the free cloudflare version if we do want the ddos protection etc. So we're stuck with api-dev.ecamp3.ch etc., and we have to be aware of the cookie problems that this brings.